### PR TITLE
Improve AdaptedRandError and other functions

### PR DESCRIPTION
**Pull request by qin-yu**
`AdaptedRandError` computes `np.unique` twice, and has the wrong dimension requirement in docstring.

### CODE REVIEW

File: pytorch3dunet/unet3d/metrics.py [hunk 1]
```diff
@@ -155,11 +149,7 @@ def _arand_err(gt, seg):
 
         per_batch_arand = []
         for _input, _target in zip(input, target):
-            n_clusters = len(np.unique(_target))
-            # skip ARand eval if there is only one label in the patch due to the zero-division error in Arand impl
-            # xxx/skimage/metrics/_adapted_rand_error.py:70: RuntimeWarning: invalid value encountered in double_scalars
-            # precision = sum_p_ij2 / sum_a2
-            if n_clusters == 1:
+            if np.all(_target == _target[0]):  # skip ARand eval if there is only one label in the patch due to zero-division
```
Comment (wolny) [inline pytorch3dunet/unet3d/metrics.py:152]: could you explain how does this logic implements checking 'if there is only one label in the patch'?

File: pytorch3dunet/unet3d/utils.py [hunk 2]
```diff
@@ -88,8 +88,7 @@ def get_logger(name, level=logging.INFO):
 
 
 def get_number_of_learnable_parameters(model):
-    model_parameters = filter(lambda p: p.requires_grad, model.parameters())
-    return sum([np.prod(p.size()) for p in model_parameters])
+    return sum(p.numel() for p in model.parameters() if p.requires_grad)
```
Comment (wolny) [inline pytorch3dunet/unet3d/utils.py:91]: nice!

**Review by wolny — CHANGES_REQUESTED** [review]
I'm not sure about the `np.all(_target == _target[0])` logic to check the number of labels in the patch. Maybe you can comment on this or make the code more readable.
